File: elixir/gigasecond/gigasecond.exs
defmodule Gigasecond do
	@doc """
	Calculate a date one billion seconds after an input date.
	"""
	@spec from({{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}) :: :calendar.datetime

	def from({{year, month, day}, {hours, minutes, seconds}} = dt) do
    secs = :calendar.datetime_to_gregorian_seconds(dt)
    :calendar.gregorian_seconds_to_datetime(secs + 1000000000)
	end
end

File: clojure/gigasecond/src/gigasecond.clj
(ns gigasecond
  (:import [java.util Calendar]))

(defn from [y m d]
  (let [c (Calendar/getInstance)]
    (.set c Calendar/YEAR y)
    (.set c Calendar/MONTH (dec m))
    (.set c Calendar/DATE d)
    (.add c Calendar/SECOND 1000000000)
    (vector (.get c Calendar/YEAR)
            (inc (.get c Calendar/MONTH))
            (.get c Calendar/DATE))))
